examples/passport-login/src/controllers/ping.controller.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/example-passport-login
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Request, RestBindings, get, ResponseObject} from '@loopback/rest';
+import {inject} from '@loopback/context';
+import {authenticate} from '@loopback/authentication';
+import {SecurityBindings, UserProfile} from '@loopback/security';
+
+
+const PING_RESPONSE: ResponseObject = {
+  description: 'Ping Response',
+  content: {
+    'application/json': {
+      schema: {
+        type: 'object',
+        title: 'PingResponse',
+        properties: {
+          user: {type: 'object'},
+          headers: {
+            type: 'object',
+            properties: {
+              'Content-Type': {type: 'string'},
+            },
+            additionalProperties: true,
+          },
+        },
+      },
+    },
+  },
+};
+
+/**
+ * A simple controller to bounce back http requests
+ */
+export class PingController {
+  constructor(@inject(RestBindings.Http.REQUEST) private req: Request) {}
+
+  @authenticate('session')
```

Comment:
User should always be able to visit /ping without authentication (even if logged in with an auth strategy)

Let's leave the /ping without  the `@authenticate` decorator

but have another endpoint in this controller like 

@authenticate('session')
@get(/whoAmi )     
which returns the currently logged in user.

What will the request headers show?

Reply:
+1, `/whoAmI` or `/me` prints the user info, `/ping` is usually public.

Reply:
agreed, this is done now